app/jobs/regular/emit_web_hook_event.rb

```diff
@@ -46,6 +50,22 @@ def web_hook_request
       end
 
       conn.post(headers: headers, body: body)
+
+      _, id, event_name = name.split('.')
+      web_hook_event = WebHookEvent.find(id)
+
+      case event_name
+        when 'request'
+          web_hook_event.header = params[:headers].to_s
+          web_hook_event.payload = params[:body].to_s
+        when 'response', 'error'
+          web_hook_event.response_code = params[:status].to_i
```

Comment:
`update_attributes!` will allow you to avoid all the `web_hook_event` repetition